client/state/preferences/reducer.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import has from 'lodash/has';
+import get from 'lodash/get';
+import forOwn from 'lodash/forOwn';
+/**
+ * Internal dependencies
+ */
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+import { USER_SETTING_KEY } from './actions';
```

Comment:
If we're using constants between files, might be better to create a separate file than to have `reducer.js` import from `actions.js`? e.g. `constants.js`